app/scripts/metamask-controller.js

```diff
@@ -4062,6 +4055,17 @@ export default class MetamaskController extends EventEmitter {
     this.sendUpdate();
   }
 
+  //=============================================================================
+  // REQUEST QUEUE
+  //=============================================================================
+
+  setUseRequestQueue = (value) => {
+    this.preferencesController.setUseRequestQueue(value);
+    this.selectedNetworkController.update((state) => {
+      state.perDomainNetwork = true;
```

Comment:
shouldnt this set `perDomainNetwork` to `value` too?

```suggestion
      state.perDomainNetwork = value;
```

Reply:
yes whoops